bundles/core/src/main/java/com/adobe/cq/wcm/core/components/services/link/DefaultLinkConfiguration.java

```diff
@@ -0,0 +1,77 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2022 Adobe
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.wcm.core.components.services.link;
+
+import java.util.Map;
+
+import org.apache.sling.api.resource.Resource;
+
+import com.adobe.cq.wcm.core.components.models.Page;
+
+public class DefaultLinkConfiguration implements LinkConfiguration {
```

Comment:
wouldn't it make more sense to have three different implementations with some inheritance?
PageLinkConfiguration -> ResourceLinkConfiguration -> UrlLinkConfiguration